### PR TITLE
Add an isdynamic function

### PR DESCRIPTION
**Pull request by Keno**
Queries whether an executable or library makes use of dynamic linking.

For https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/83

### CODE REVIEW

**Comment by Keno** [comment]
@staticfloat

File: src/Abstract/DynamicLink.jl [hunk 1]
```diff
@@ -156,6 +158,7 @@ function find_library(rpath::RPath, soname::AbstractString)
     return soname
 end
 
+@mustimplement isdynamic(oh::ObjectHandle)
```
Comment (staticfloat) [inline src/Abstract/DynamicLink.jl]: Put this in `ObjectHandle.jl`, like right [here](https://github.com/staticfloat/ObjectFile.jl/blob/98d7b327448456df024ab87243520f972c715e16/src/Abstract/ObjectHandle.jl#L216).  Add a docstring to the abstract definition, similar to `islibrary()`, add it to the list of methods [here](https://github.com/staticfloat/ObjectFile.jl/blob/98d7b327448456df024ab87243520f972c715e16/src/Abstract/ObjectHandle.jl#L46-L56).

**Comment by Keno** [comment]
done

**Comment by staticfloat** [comment]
Okay, add a test and this is good to go.

**Comment by Keno** [comment]
also done

**Pull request merged**